### PR TITLE
Add many to many indigeneity checkboxes

### PR DESCRIPTION
**Pull request by jonathan-longe**
Create database migrations for a new "indigeneity_types" and "student_indigeneity" tables

Admin users can create and modify indigeneity types on the maintenance page:
Screenshot (87)

When adding or editing a student's profile, users have the ability to select zero, one or more indigeneity checkboxes as appropriate:
Screenshot (88)

### CODE REVIEW

File: Modules/Twp/Database/Migrations/2024_02_08_232612_add_student_indigeneity_table.php
Comment (kmarshgov) [inline Modules/Twp/Database/Migrations/2024_02_08_232612_add_student_indigeneity_table.php:1]: You don't need id and timestamps for a pivot table

File: Modules/Twp/Resources/assets/js/Components/StudentEditStudentTab.vue [hunk 1]
```diff
@@ -60,27 +60,27 @@ tr {
...
-                </BreezeSelect>
-            </div>
-            <div class="col-md-4">
+
+            <div class="col-md-6">
                 <BreezeLabel for="inputAge" class="form-label" value="Age" />
                 <BreezeInput type="number" class="form-control bg-light" id="inputAge" v-model="editForm.age" readonly="readonly" disabled />
             </div>
 
+            <div class="col-md-12">
```
Comment (kmarshgov) [inline Modules/Twp/Resources/assets/js/Components/StudentEditStudentTab.vue:76]: This block should be shown only if the student is a citizen

File: Modules/Twp/Resources/assets/js/Pages/Students.vue [hunk 2]
```diff
@@ -108,30 +108,29 @@
...
+
+                                    <div class="col-md-6">
                                         <BreezeLabel for="inputGender" class="form-label" value="Gender" />
                                         <BreezeSelect class="form-select" id="inputGender" v-model="newStudentForm.gender">
                                             <option value="M">Male</option>
                                             <option value="F">Female</option>
                                             <option value="O">Other</option>
                                         </BreezeSelect>
                                     </div>
+                                    <div class="col-md-12">
```
Comment (kmarshgov) [inline Modules/Twp/Resources/assets/js/Pages/Students.vue:127]: This block should be shown only if the student is a citizen

File: Modules/Twp/Entities/IndigeneityType.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Modules\Twp\Entities;
+
+class IndigeneityType extends ModuleModel
+{
+
+    protected $fillable = ['title', 'is_active'];
+    protected $table = 'indigeneity_types';
```
Comment (kmarshgov) [inline Modules/Twp/Entities/IndigeneityType.php:9]: no need for $table in here.

**Review by kmarshgov — CHANGES_REQUESTED** [review]
indigeneity is used in ReportController. Pull down the updates and then append indigeneitytype to the student model so we can view it as a relationship under students.

File: Modules/Twp/Http/Controllers/ReportController.php [hunk 4]
```diff
@@ -84,7 +84,9 @@ public function fetchReport(Request $request, $type): \Illuminate\Http\JsonRespo
     public function students(Request $request)
     {
         return response()->json(Student::select('id', 'first_name', 'last_name', 'sin', 'birth_date',
-            'address', 'pen', 'email', 'gender', 'citizenship', 'bc_resident', 'indigeneity', 'comment')->get(), 200);
+            'address', 'pen', 'email', 'gender', 'citizenship', 'bc_resident', 'comment')
+            ->with('indigeneity')
+            ->get(), 200);
```
Comment (jonathan-longe) [inline Modules/Twp/Http/Controllers/ReportController.php:89]: @kmarshgov like this?

Comment (kmarshgov) [reply Modules/Twp/Http/Controllers/ReportController.php:89]: You got it.

File: Modules/Twp/Entities/IndigeneityType.php [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+namespace Modules\Twp\Entities;
+
+class IndigeneityType extends ModuleModel
+{
+
+    protected $fillable = ['title', 'is_active'];
+
+    public function students()
+    {
+        return $this->belongsToMany(
```
Comment (kmarshgov) [inline Modules/Twp/Entities/IndigeneityType.php:12]: This should be enough        return $this->belongsToMany('Modules\Twp\Entities\Student', 'indigeneity_student');

File: Modules/Twp/Entities/Student.php [hunk 6]
```diff
@@ -37,6 +37,13 @@ public function application()
         return $this->hasOne('Modules\Twp\Entities\Application', 'student_id', 'id');
     }
 
+    public function indigeneity()
+    {
+        return $this->belongsToMany(
```
Comment (kmarshgov) [inline Modules/Twp/Entities/Student.php:42]: return $this->belongsToMany('Modules\Twp\Entities\IndigeneityType', 'indigeneity_student');

File: Modules/Twp/Database/Migrations/2024_02_08_232612_add_student_indigeneity_table.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::connection(env('DB_DATABASE_TWP'))->create('indigeneity_student', function (Blueprint $table) {
+            $table->foreignId('student_id')->constrained('students')->onDelete('cascade');
+            $table->foreignId('indigeneity_id')->constrained('indigeneity_types')->onDelete('cascade');
```
Comment (kmarshgov) [inline Modules/Twp/Database/Migrations/2024_02_08_232612_add_student_indigeneity_table.php:18]: rename indigeneity_id to indigeneity_type_id

Comment (kmarshgov) [reply Modules/Twp/Database/Migrations/2024_02_08_232612_add_student_indigeneity_table.php:18]: rename the file to ... create_indigeneity_type_student_table
rename the table to indigeneity_type_student and it's reference in the models

**Review by kmarshgov — APPROVED** [review]

**Pull request merged**